packages/react-dev-utils/FlowTypecheckPlugin.js

```diff
@@ -0,0 +1,243 @@
+      stdout = Buffer.concat([stdout, chunk]);
+    });
+    oneTimeProcess.stderr.on('data', chunk => {
+      stderr = Buffer.concat([stderr, chunk]);
+    });
+    oneTimeProcess.on('error', error => reject(error));
+    oneTimeProcess.on('exit', code => {
+      switch (code) {
+        case 0:
+          return resolve(stdout);
+        default:
+          return reject(new Error(
+            Buffer.concat([stdout, stderr]).toString()
+          ));
+      }
+    });
+  });
+}
+
+function writeFileIfDoesNotExist(path, data) {
+  return new Promise((resolve, reject) => {
+    fs.exists(path, exists => {
+      if (!exists) {
+        fs.writeFile(path, data, err => {
+          if (err) {
+            return reject(err);
+          }
+          resolve(true);
+        });
+      } else {
+        resolve(false);
+      }
+    });
+  });
+}
+
+function writeInFileIfNotPresent(path, contentToAssert, contentToAppend) {
+  return new Promise((resolve, reject) => {
+    fs.exists(path, exists => {
+      if (!exists) {
+        fs.writeFile(path, contentToAppend, err => {
+          if (err) {
+            return reject(err);
+          }
+          resolve(true);
+        });
+      } else {
+        fs.readFile(path, (err, existingContent) => {
+          if (err) {
+            return reject(err);
+          }
+          if (existingContent.indexOf(contentToAssert) < 0) {
+            fs.appendFile(path, contentToAppend, err => {
+              if (err) {
+                return reject(err);
+              }
+              resolve(true);
+            });
+          } else {
+            resolve(false);
+          }
+        });
+      }
+    });
+  });
+}
+
+function getFlowVersion(options) {
+  return execOneTime(
+    (options || {}).global ? "flow" : flowBinPath,
+    ['version', '--json']
+  )
+  .then(rawData => JSON.parse(rawData))
+  .then(versionData => versionData.semver);
+}
+
+function initializeFlow(projectPath, flowconfig, otherFlowTypedDefs) {
+  const flowconfigPath = path.join(projectPath, '.flowconfig');
+  const gitignorePath = path.join(projectPath, '.gitignore');
+  return getFlowVersion().then(localVersion => 
+    getFlowVersion({global: true}).catch(() => localVersion)
+    .then(globalVersion =>
+      globalVersion !== localVersion ?
+        Promise.reject(new Error(
+          'Flow integration was disabled because the global Flow version does not match.\n' +
+          'You may either remove the global Flow installation or install a compatible version:\n' +
+          '  ' + chalk.cyan('npm') + ' install -g flow-bin@' + localVersion
+        )) :
+        localVersion
+    )
+  )
+  .then(localVersion => Promise.all([
+    writeFileIfDoesNotExist(flowconfigPath, flowconfig.join('\n'))
+    .then(wroteFlowconfig => wroteFlowconfig ?
+      writeInFileIfNotPresent(gitignorePath, 'flow-typed', 'flow-typed/npm') :
+      false
+    ),
+    execOneTime(
+      flowTypedPath,
```

Comment:
What is the alternative to using `flow-typed`? AFAIK it is explicitly encouraged by Flow team as the official way to add typings for dependencies, and in some cases it is intentional that you can [modify it by hand](https://flowtype.org/docs/third-party.html#writing-custom-library-definition-files). I don't think hiding it would be appropriate.